oauth-common/src/main/java/io/strimzi/kafka/oauth/validator/JWTSignatureValidator.java

```diff
@@ -254,16 +228,34 @@ private PublicKey getKeyUnlessStale(String id) {
 
     private void fetchKeys() {
         try {
-            JSONWebKeySet jwks = HttpUtil.get(keysUri, socketFactory, hostnameVerifier, null, JSONWebKeySet.class);
-            Map<String, PublicKey> newCache = JWKSUtils.getKeysForUse(jwks, JWK.Use.SIG);
+            Map<String, PublicKey> newCache = new HashMap<>();
+            JWKSet jwks = JWKSet.parse(HttpUtil.get(keysUri, socketFactory, hostnameVerifier, null, String.class));
+
+            for (JWK jwk: jwks.getKeys()) {
+                if (jwk.getKeyUse().equals(KeyUse.SIGNATURE)) {
+
+                    PublicKey publicKey;
+
+                    if (jwk instanceof ECKey) {
+                        publicKey = ((ECKey) jwk).toPublicKey();
+                    } else if (jwk instanceof RSAKey) {
+                        publicKey = ((RSAKey) jwk).toPublicKey();
+                    } else {
+                        log.warn("Unsupported JWK key type: " + jwk.getKeyType());
+                        continue;
+                    }
+                    newCache.put(jwk.getKeyID(), publicKey);
+                }
+            }
+
             newCache = Collections.unmodifiableMap(newCache);
             if (!cache.equals(newCache)) {
                 log.info("JWKS keys change detected. Keys updated.");
                 oldCache = cache;
                 cache = newCache;
             }
             lastFetchTime = System.currentTimeMillis();
-        } catch (Exception ex) {
+        } catch (Throwable ex) {
```

Comment:
What non-Exception can be thrown here now?

Reply:
There are no additional exceptions / errors that could be thrown here. I just noticed, that non-exceptions can make it through uncaught, potentially avoiding the log or making the log entry without the information about the SSO endpoint url that was affected.